developer_docs/codegen_stage.md

```diff
@@ -0,0 +1,281 @@
+On the other hand implementations written directly in the generated code can decrease readability of the code (try translating the print of an array for an example).
+
+In the future we hope to add an additional option to avoid some of these problems where an implementation of common low-level functions (e.g. `sum`) is written in Python and Pyccel is used to provide the language-specific equivalent of this function with the required argument types.
+This option requires some discussion amongst the senior developers and some additional functions (e.g. `reshape`/`ravel`/`flatten`).
+
+## Fortran-specific problems
+
+### 0-based arrays
+
+Unlike in the majority of coding languages, Fortran array indexes do not have to start from 0.
+In fact, the default start index is 1.
+As a result this needs to be taken into consideration when writing code.
+
+There are two possibilities to handle this; either we can modify every index by adding 1, or we can define our arrays in Fortran with syntax which ensures that the indexes start from 0.
+In Pyccel we have chosen the latter option.
+It is therefore very important to think about this when allocating arrays or pointers.
+
+### Ordering
+
+Different languages handle multi-dimensional arrays in different ways.
+This means that optimal indexing is not written in the same way in different languages.
+Specifically C uses row-major indexing, while Fortran uses column-major indexing.
+Python allows both indexing conventions, but it defaults to C ordering.
+As a result this must be taken into consideration when translating to Fortran in order to preserve efficient code.
+In order to avoid unnecessary copying of array arguments Pyccel handles this difference via the printing.
+This solution is quite complex so it has its own documentation.
+
+Please see the [Order Docs](./order_docs.md) for more details.
+
+## Python-specific problems
+
+Translating Python to Python is theoretically very simple and serves mostly to test our AST (although this functionality is also leveraged by packages such as PSYDAC).
+However there are occasionally complications as the AST is more complex than necessary to describe a python function.
+
+### Temporary variables
+
+Pyccel occasionally introduces temporary variables to help express statements in low-level languages.
+Where the language permits, we aim to not print these variables.
+As they did not exist in the original Python file it should therefore be possible to completely avoid these variables when printing Python code.
+The variables can be identified by the `is_temp` property.
+
+An example of this is return variables.
+In Pyccel the objects returned by a function are always saved into variables.
+This is useful as it provides a Variable to help define the function signature.
```

Comment:
Missing quotes for inline code: `Variables`

Reply:
Fixed a88e2512